mkosi

```diff
@@ -415,8 +415,22 @@ def reuse_cache_image(args, workspace, run_build_script, for_cache):
         with source:
             f = tempfile.NamedTemporaryFile(dir = os.path.dirname(args.output), prefix='.mkosi-')
             output.append(f)
+
+            # So on one hand we want CoW off, since this stuff will
+            # have a lot of random write accesses. On the other we
+            # want the copy to be snappy, hence we do want CoW. Let's
+            # ask for both, and let the kernel figure things out:
+            # let's turn off CoW on the file, but start with a CoW
+            # copy. On btrfs that works: the initial copy is made as
+            # CoW but later changes do not result in CoW anymore.
```

Comment:
I found it a bit confusing that this long block of documentation precedes just a single call to `_reflink` – where is all this “turn off CoW but start with CoW” logic hiding? :) IIUC that’s just the behavior of the `FICLONE` ioctl, so I think some of this documentation should be placed on `_reflink` instead.

Reply:
not quite. we first invoke "disable_cow()" and then follow up with "_reflink()" right after. it's that combination of first turning off cow, and then doing a reflink (which is the mother of all cow concepts) after all that deserves the comment.

Reply:
Wait, how the fuck did I not see that `disable_cow` call? :D

![Ben Mendelsohn playing Orson Krennic in “Rogue One: A Star Wars Story”, in the scene on Scarif just after the rebel attack starts, yelling at the officers surrounding him: “Are we blind? Deploy the garrison!”](https://user-images.githubusercontent.com/2346599/34632679-248a2178-f277-11e7-9d2d-ee9670161f65.png "Are we blind?")